drivers/hwinfo/hwinfo_cc13xx_cc26xx.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2023 Piotr Dymacz
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <string.h>
+#include <zephyr/drivers/hwinfo.h>
+#include <zephyr/sys/byteorder.h>
+
+#include <driverlib/sys_ctrl.h>
+#include <inc/hw_ccfg.h>
+#include <inc/hw_fcfg1.h>
+#include <inc/hw_memmap.h>
+
+#ifdef CONFIG_HWINFO_CC13XX_CC26XX_USE_BLE_MAC
+#define CC13XX_CC26XX_DEVID_SIZE	6
+#else
+#define CC13XX_CC26XX_DEVID_SIZE	8
+#endif
+
+ssize_t z_impl_hwinfo_get_device_id(uint8_t *buffer, size_t length)
+{
+	uint8_t *mac;
+
+#ifdef CONFIG_HWINFO_CC13XX_CC26XX_USE_BLE_MAC
```

Comment:
Unless I overlook some technical reason why we need defines, I believe there are reasons to prefer the following pattern:

```suggestion
if (IS_ENABLED(CONFIG_HWINFO_CC13XX_CC26XX_USE_BLE_MAC) || sys_read32(CCFG_BASE + CCFG_O_IEEE_BLE_0) == 0xFFFFFFFF || sys_read32(CCFG_BASE + CCFG_O_IEEE_BLE_1) == 0xFFFFFFFF) {
...
}
```

* easier to read and maintain
* ensures that all code is compiled independently of compile time config options (avoids bit rot).
* In your specific case it also removes redundant code

The compiler will optimize unused code away so that the end result is not wasting more flash - at least if you put the check on the constant value first in the expression. You can easily check this in the resulting assembly code if you look at the result through some debugger frontend.

Reply:
I switched to `IS_ENABLED()` but in a slightly different way ('always use factory defaults' support).
Thanks for the suggestion!